pkg/kubelet/config/file_linux_test.go

```diff
@@ -107,69 +105,78 @@ func TestReadPodsFromFileExistAlready(t *testing.T) {
 	}
 }
 
+type WatchType int
+type FileType int
+
+const (
+	WATCH_DIR WatchType = iota
+	WATCH_FILE
+
+	FILE FileType = iota
+	SYMLINK
+)
+
+var (
+	watchFileAddedTesters = map[WatchType]map[FileType]func(*testing.T){
```

Comment:
You don't need all this extra delegation machinery. 
You only need two test functions:
- `TestWatchFileAdded`
- `TestWatchFileChanged`

And just pass the full boolean matrix in each.